File: 백준/Gold/1407. 2로 몇 번 나누어질까/2로 몇 번 나누어질까.cc
#include <iostream>
using namespace std;

int main() {
  long long a, b;
  cin>>a>>b;

  long long x = 1;
  while(x * 2 <= b) {
    x *= 2;
  }
  
  long long count = 0, sum = 0, cur;
  while(x > 0) {
    cur = b / x - a / x + (a % x == 0) - count;
    count += cur;
    sum += x * cur;
    x /= 2;
  } 
  cout<<sum<<'\n';
  return 0;
}

File: 백준/Gold/1407. 2로 몇 번 나누어질까/2로 몇 번 나누어질까.js
const fs = require("fs");
const [a, b] = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(" ")
  .map(Number);

function solution(a, b) {
  let prev = 1n,
    sum = 0n;
  for (let x = 1; x <= b; x *= 2) {
    const cur = Math.floor(b / x) - Math.ceil(a / x) + 1;
    sum += BigInt(cur) * prev;
    prev = BigInt(x);
  }
  return sum.toString();
}

console.log(solution(a, b));
